anyway/views/news_flash/api.py

```diff
@@ -199,19 +200,75 @@ def gen_news_flash_query(
 
     return query
 
+def filter_by_road_number(query: Query, value: float, **kwargs) -> Query:
+    return query.filter(NewsFlash.road1 == value)
+
+def filter_by_source(query: Query, value: Optional[List[BE_CONST.Source]], **kwargs) -> Query:
+    if value is None:
+        return query
+    sources = set([source.value for source in value])
+    critical = BE_CONST.Source.CRITICAL.value
+    if critical in sources:
+        query = query.filter(NewsFlash.critical==True)
+        sources.remove(critical)
+        if len(sources) == 0:
+            return query
+    return query.filter(NewsFlash.source.in_(list(sources)))
+
+def filter_by_dates(query: Query, value: Optional[datetime.datetime], **kwargs) -> Query:
+    start_date, end_date = value, kwargs.get('end_date')
+    if start_date:
+        if end_date:
+            query = query.filter(start_date <= NewsFlash.date <= end_date)
+        else:
+            query = query.filter(start_date <= NewsFlash.date)
+    elif end_date:
+        query = query.filter(NewsFlash.date <= end_date)
+    return query
+
+def filter_by_resolutions(query, value: List[str], **kwargs):
+    ands = []
+    if "suburban_road" in value:
+        ands.append(
+            and_(
+                NewsFlash.resolution == BE_CONST.ResolutionCategories.SUBURBAN_ROAD.value,
+                NewsFlash.road_segment_name != None,
+            )
+        )
+    if "street" in value:
+        ands.append(
+            and_(
+                NewsFlash.resolution == BE_CONST.ResolutionCategories.STREET.value,
+                NewsFlash.street1_hebrew != None,
+            )
+        )
```

Comment:
Hi, I think it is the role of the code that generates a news-flash to validate that the entered item is legal. A legal value means that if it has a given resolution then the required fields for this resolution are not empty.
Not relying on it means we will have many places in our code that performs these same tests. These tests change when we add resolutions, and when required fields of resolution changes (when we better understand the behavior of that resolution). If our current news-flash data may not be valid or clean, I suggest to set a periodic process that will scan and fix it. 
It will be easier to maintain the code if it works with  `resolution_dict` and `SUPPORTED_RESOLUTIONS`, and avoids using specific resolutions and fields.
Currently the resolution-specifc code resides in `request_params.py` file. Thinking of it, we should create a `Resolution` set of classes that will encapsulate all the per-resolution knowledge.